app/bundles/FormBundle/Model/FormModel.php

```diff
@@ -788,24 +788,31 @@ public function populateValuesWithGetParameters(Form $form, &$formHtml)
     public function populateValuesWithLead(Form $form, &$formHtml)
     {
         $formName = $form->generateFormName();
-        $lead     = $this->leadModel->getCurrentLead();
+        $fields   = $form->getFields();
+        /** @var \Mautic\FormBundle\Entity\Field $f */
+        foreach ($fields as $key=> $f) {
+            $leadField  = $f->getLeadField();
+            $isAutoFill = $f->getIsAutoFill();
+
+            if (!isset($leadField) || !$isAutoFill) {
+                unset($fields[$key]);
+            }
+        }
+        // no fields for populate
+        if (!count($fields)) {
+            return;
+        }
 
+        $lead     = $this->leadModel->getCurrentLead();
         if (!$lead instanceof Lead) {
             return;
         }
 
-        $fields = $form->getFields();
-        /** @var \Mautic\FormBundle\Entity\Field $f */
         foreach ($fields as $f) {
-            $leadField  = $f->getLeadField();
-            $isAutoFill = $f->getIsAutoFill();
-
-            if (isset($leadField) && $isAutoFill) {
-                $value = $lead->getFieldValue($leadField);
+            $value = $lead->getFieldValue($leadField);
```

Comment:
Shouldn't this be `$f`? Although might be better to rename `$f` to something more descriptive and update in line 815 as well.

Reply:
That has been addressed also